powersoftau/src/batched_accumulator.rs

```diff
@@ -872,6 +858,278 @@ mod tests {
         serialize_accumulator_curve::<SW6>(UseCompression::No, 2, 2);
     }
 
+    #[test]
+    fn generate_initial_test() {
+        generate_initial_test_curve::<Bls12_377>(4, 4, UseCompression::Yes);
+        generate_initial_test_curve::<Bls12_377>(4, 4, UseCompression::No);
+    }
+
+    #[test]
+    fn test_contribute() {
+        // receive a compressed/uncompressed input, contribute to it and produce
+        // a compressed/decompressed output
+        test_contribute_curve::<Bls12_377>(2, 2, UseCompression::Yes, UseCompression::Yes);
+        test_contribute_curve::<Bls12_377>(2, 2, UseCompression::No, UseCompression::Yes);
+        test_contribute_curve::<Bls12_377>(2, 2, UseCompression::Yes, UseCompression::No);
+        test_contribute_curve::<Bls12_377>(3, 4, UseCompression::No, UseCompression::No);
+        test_contribute_curve::<Bls12_377>(6, 64, UseCompression::No, UseCompression::No);
+        // works even if the batch is larger than the powers
+        test_contribute_curve::<Bls12_377>(6, 128, UseCompression::No, UseCompression::Yes);
```

Comment:
Note: the current implementation from Matter failed in this test case. I have a follow up re-implementation which addresses that.